lib/Driver/Driver.cpp

```diff
@@ -663,13 +692,9 @@ Driver::buildCompilation(const ToolChain &TC,
     ArgList->hasArg(options::OPT_driver_show_incremental);
   bool ShowJobLifecycle =
     ArgList->hasArg(options::OPT_driver_show_job_lifecycle);
-  if (const Arg *A = ArgList->getLastArg(options::OPT_driver_batch_seed)) {
-    if (StringRef(A->getValue()).getAsInteger(10, DriverBatchSeed)) {
-      Diags.diagnose(SourceLoc(), diag::error_invalid_arg_value,
-                     A->getAsString(*ArgList), A->getValue());
-    }
-  }
-  DriverForceOneBatchRepartition =
+  unsigned DriverBatchSeed = getDriverBatchSeed(ArgList, Diags);
+  Optional<unsigned> DriverBatchCount = getDriverBatchCount(ArgList, Diags);
```

Comment:
Nice! Thanks for factoring; I think it helps readability. When I come back to https://github.com/apple/swift/pull/16762 (waiting for @jrose-apple ), I will move these computations closer to their uses and constify it.